kernel/alarm.c

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (c) 2019 Peter Bigot Consulting, LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <logging/log.h>
+
+extern struct k_alarm _k_alarm_list_start[];
+extern struct k_alarm _k_alarm_list_end[];
+
+static struct k_spinlock lock;
+static sys_dlist_t scheduled_list = SYS_DLIST_STATIC_INIT(&scheduled_list);
+static sys_dlist_t ready_list = SYS_DLIST_STATIC_INIT(&ready_list);
+
+/** Return a pointer to the k_alarm that contains the node if it has a
+ * deadline at or before the provided when.  Return a null pointer if
+ * the k_alarm is scheduled strictly after the provided when.
+ */
+static ALWAYS_INLINE struct k_alarm *alarm_due_by(sys_dnode_t *node,
+						  uint32_t when)
+{
+	struct k_alarm *cap = CONTAINER_OF(node, struct k_alarm, node);
+
+	if ((s32_t)(cap->deadline - when) > 0) {
+		return 0;
+	}
+	return cap;
+}
+
+/** Add alarm to the provided list before the first alarm that has a
+ * later deadline, or at the end if necessary.
+ *
+ * Return @true if the head of list changed as a result of this
+ * addition.
+ */
+static bool link_alarm(sys_dlist_t *list,
+		       struct k_alarm *alarm,
+		       u32_t now)
+{
+	sys_dnode_t *node = &alarm->node;
+	sys_dnode_t *cp = sys_dlist_peek_head(list);
+
+	while (cp) {
+		if (!alarm_due_by(cp, alarm->deadline)) {
+			break;
+		}
+		cp = sys_dlist_peek_next_no_check(list, cp);
```

Comment:
aerate please, no need to coalesce } and next lines (unless it's another '}')

Apply that everywhere.